zkevm-circuits/src/pi_circuit.rs

```diff
@@ -1092,19 +1285,81 @@ impl<F: Field> PiCircuitConfig<F> {
             )?;
         }
 
+        //////////////////////////////////////////////
+        ///////  assign l1 block hashes count ////////
+        //////////////////////////////////////////////
+        // println!("public_data.prev_last_applied_l1_block: {}", public_data.prev_last_applied_l1_block);
+        let num_all_l1_block_hashes = public_data.get_num_all_l1_block_hashes() as u64;
+        let mut l1_block_hashes_count = num_all_l1_block_hashes;
+        // The first time there are l1 block hashes, the prev_last_applied_l1_block is 0.
+        if public_data.prev_last_applied_l1_block != 0 {
+            assert!(
+                public_data.last_applied_l1_block >= public_data.prev_last_applied_l1_block,
+                "last_applied_l1_block should be greater or equal than prev_last_applied_l1_block");
+            l1_block_hashes_count =
+                public_data.last_applied_l1_block - public_data.prev_last_applied_l1_block;
+        }
```

Comment:
shouldn't this simply be:
```rust
let l1_block_hashes_count = public_data.last_applied_l1_block - public_data.prev_last_applied_l1_block;
```
?

Reply:
the first time we have l1 block hashes in a block the `prev_last_applied_l1_block` will be 0 and then we don't want  to have `let l1_block_hashes_count = public_data.last_applied_l1_block - public_data.prev_last_applied_l1_block;` because it will be an incorrect value, so in that case we just have an unuseful constraint that `num_all_l1_block_hashes == num_all_l1_block_hashes`